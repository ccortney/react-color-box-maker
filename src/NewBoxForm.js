import {useState} from "react";

const NewBoxForm = ({addBox}) => {
    const initial_state = {
        color: "#000000", 
        side: ""
    }

    const [formData, setFormData] = useState(initial_state);

    const handleChange = (e) => {
        const {name, value} = e.target;
        setFormData(formData => ({...formData, [name]: value}))
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        addBox(formData);
        setFormData(initial_state)
    }

    return (
        <form onSubmit={handleSubmit}>
            <label htmlFor="color">Choose a Color</label>
            <input
                type="color"
                id="color"
                name="color"
                value={formData.color}
                onChange={handleChange}
            />
            <label htmlFor="side">Enter a Side Length</label>
            <input
                type="text"
                id="side"
                name="side"
                value={formData.side}
                size="5"
                minLength="1"
                maxLength="3"
                onChange={handleChange}
            />
            <button>Create Box</button>
        </form>
    )
}

export default NewBoxForm;