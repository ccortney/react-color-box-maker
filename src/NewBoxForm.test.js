import { render, screen } from '@testing-library/react';
import NewBoxForm from './NewBoxForm';

// smoke test
test('renders without crashing', () => {
  render(<NewBoxForm />);
})

// snapshot test
test ('matches snapshot', () => {
  const {asFragment} = render(<NewBoxForm/>);
  expect(asFragment()).toMatchSnapshot();
})