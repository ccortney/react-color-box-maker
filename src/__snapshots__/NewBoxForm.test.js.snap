// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="color"
    >
      Choose a Color
    </label>
    <input
      id="color"
      name="color"
      type="color"
      value="#000000"
    />
    <label
      for="side"
    >
      Enter a Side Length
    </label>
    <input
      id="side"
      maxlength="3"
      minlength="1"
      name="side"
      size="5"
      type="text"
      value=""
    />
    <button>
      Create Box
    </button>
  </form>
</DocumentFragment>
`;
